### PR TITLE
Voting Routes [POST]

### PR DESCRIPTION
**Pull request by lgv-0**
# Description

Implements the second half (submission) of the voting component

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge

### CODE REVIEW

**Review by crutledgedev — APPROVED** [review]

**Review by LeoSanchez89 — APPROVED** [review]

**Comment by lgv-0** [comment]
[Endpoints for independent voting](https://trello.com/c/TApCphGH/15-endpoints-for-independent-voting)

File: src/routes/voting/voting.Routes.ts [hunk 1]
```diff
@@ -1,86 +1,126 @@
...
-            });
+        let User = req.user as Child;
+        let ChildMatch = await FindMatchByUID(User.id, User.cohort.week);
 
-            if (temp) LowestNeeded = iter;
+        //Get a random versus matchup that is NOT from this childs match
+        let response = await randomIgnoring(ChildMatch);
 
-            iter++;
+        if (!response) {
```
Comment (ryan-hamblin) [inline src/routes/voting/voting.Routes.ts:22]: I'm wondering here if there is a way to actually determine the error and return that? Is this a restriction from the DB that you're using.

File: src/routes/voting/voting.Routes.ts [hunk 2]
```diff
@@ -1,86 +1,126 @@
...
-            .where({ match: Not(ignore.id), votes: LowestNeeded })
-            .orderBy('RANDOM()')
-            .limit(1)
-            .getOne();
+        let Build = {} as any;
+
+        Build.matchupID = response.id;
+
+        if (response.story) {
+            //If story, grab them and relay them back
```
Comment (ryan-hamblin) [inline src/routes/voting/voting.Routes.ts:33]: Love these comments! Thank you for organizing your code so neatly.

File: src/routes/voting/votingRoutes.functions.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { connection } from '../../util/typeorm-connection';
+import { Versus } from '../../database/entity/Versus';
+import { Matches } from '../../database/entity';
+
+export async function randomIgnoring(ignore: Matches) {
+    const manager = getManager(connection());
+    const VersusRepo = manager.getRepository(Versus);
+
+    try {
+        let iter = 0;
```
Comment (ryan-hamblin) [inline src/routes/voting/votingRoutes.functions.ts:11]: I'm not the biggest fan of abbreviated variable names that don't semantically tell me what whole story is. Consider revision.

**Review by ryan-hamblin — COMMENTED** [review]
This PR looks great! I'd love to see a lot more back and forth in the comments from those that reviewed. It looks to me that you're committing code often (LOVE THAT) and your commits are well written and atomic. I also see that this was a complete single feature, and it only touched 2 files! That's great! This is the type of work that you'll be doing day in and day out at your job. 

Getting a bit nit-pick I'd recommend revising some variable names in the future to ensure that future devs can easily understand your work. Your code commenting was AMAZING. I know what most of your code is doing by just reading the comments you left about it. I'd recommend writing more! Great job!

**Pull request merged**